### DISCUSSION TITLE
Registering template locations recursively: 👍/👎?

### DISCUSSION THREAD

**Discussion by cbirdsong** [timber/timber#2868] [Q&A]
One of the template locations we register with Timber is for icon SVG files so we can just do `{% include "whatever.svg" %}`. It often becomes cleaner to group those into folders for different sources and use cases: 

```php
Timber::$dirname = [
	"icons",
	"icons/social",
	"icons/brand",
	// etc, etc
]
```
Doing it this way means we have to go and update those directories anytime we reorganize the icons, something we could sidestep registering these with a glob:

```php
$template_locations = ["/icons/"];

$directory_search = glob(get_stylesheet_directory() . "/icons/*", GLOB_ONLYDIR);
foreach ($directory_search as $key => $dir) {
	$dir_relative = str_replace(get_stylesheet_directory(), "", $dir);

	$directories[$dir] = $name;
}

Timber::$dirname = $template_locations;
```

Is there any reason to avoid doing that when registering directories with Timber/Twig? #1612 made me think it might cause performance issues, but: 
- it was asked in 2017
- the example uses recursive iterators and registers a _lot_ more directories than we ever would
- the issue was never followed up on until it was recently closed without comment

_(also open to any other suggestions for how to deal with this scenario more robustly)_

**Comment by Levdbas** [comment]
Hi @cbirdsong ,

But why don't you just register icons and then use:

```twig
{% include "socials/whatever.svg" %}
```

or create your own twig registered function where you can do some custom looping if required? This way you could even mark icons as deprecated, changed etc.

So in twig you would just use
```twig
{{ get_icon('socials/whatever') }}
```

Hope that sparks some ideas!

**Reply by cbirdsong** [reply]
You know, either of those are definitely a more straightforward/less overthought way to deal with this. 😂 I will probably just go with one of them, though I am still curious if there are performance implications around the number of template locations. Thanks!

**Reply by Levdbas** [reply]
Yes there are most definitely performance implications. Every directory lookup is an i/o call if the file exists, if not it tries the next one. So adding many directories is not recommended.
